centreon/src/Core/Notification/Application/Converter/NotificationHostEventConverter.php

```diff
@@ -0,0 +1,156 @@
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Notification\Application\Converter;
+
+use Core\Notification\Domain\Model\NotificationHostEvent;
+
+class NotificationHostEventConverter
+{
+    private const CASE_UP_AS_BIT = 0b001;
+    private const CASE_DOWN_AS_BIT = 0b010;
+    private const CASE_UNREACHABLE_AS_BIT = 0b100;
+
+    private const MAX_BITFLAG = 0b111;
+
+    private const CASE_UP_AS_STR = 'o';
+    private const CASE_DOWN_AS_STR = 'd';
+    private const CASE_UNREACHABLE_AS_STR = 'u';
+
+    /**
+     * Convert an array of NotificationHostEvent to a string.
+     * ex: [NotificationHostEvent::Down, NotificationHostEvent::Unreachable] => 'd,u'
+     *
+     * @param NotificationHostEvent[] $events
+     *
+     * @return string
+     */
+    public static function toString(array $events): string
+    {
+        $eventsAsBitFlag = [];
+        foreach ($events as $event) {
+            $eventsAsBitFlag[] = match ($event) {
+                NotificationHostEvent::Up => self::CASE_UP_AS_STR,
+                NotificationHostEvent::Down => self::CASE_DOWN_AS_STR,
+                NotificationHostEvent::Unreachable => self::CASE_UNREACHABLE_AS_STR,
+            };
+        }
+
+        return implode(',', array_unique($eventsAsBitFlag));
+    }
+
+    /**
+     * Convert a string to an array of NotificationHostEvent.
+     * ex: 'd,u' => [NotificationHostEvent::Down, NotificationHostEvent::Unreachable]
+     *
+     * @param string $legacyStr
+     *
+     * @return NotificationHostEvent[]
+     */
+    public static function fromString(string $legacyStr): array
+    {
+        if ($legacyStr === '') {
+            return [];
+        }
+
+        $legacyValues = explode(',', $legacyStr);
+        $legacyValues = array_unique(array_map(trim(...), $legacyValues));
+        $events = [];
+        foreach ($legacyValues as $value) {
+            $events[] = match ($value) {
+                self::CASE_UP_AS_STR => NotificationHostEvent::Up,
+                self::CASE_DOWN_AS_STR => NotificationHostEvent::Down,
+                self::CASE_UNREACHABLE_AS_STR => NotificationHostEvent::Unreachable,
+            };
+        }
+
+        return $events;
+    }
+
+    /**
+     * Convert a NotificationHostEvent into bitmask.
+     *
+     * @param NotificationHostEvent $event
+     *
+     * @return int
+     */
+    public static function toBit(NotificationHostEvent $event): int
+    {
+        return match ($event) {
+            NotificationHostEvent::Up => self::CASE_UP_AS_BIT,
+            NotificationHostEvent::Down => self::CASE_DOWN_AS_BIT,
+            NotificationHostEvent::Unreachable => self::CASE_UNREACHABLE_AS_BIT,
+        };
+    }
+
+    /**
+     * Convert a bitmask into an array of NotificationHostEvent.
+     *
+     * @param int $bitmask
+     *
+     * @throws \Throwable
+     *
+     * @return NotificationHostEvent[]
+     */
+    public static function fromBitFlag(int $bitFlag): array
```

Comment:
It is more a list of flags : I would use the `bitFlags` wording ?
And be careful : phpdoc is not consistent.
```suggestion
     * @param int $bitFlags
     *
     * @throws \Throwable
     *
     * @return NotificationHostEvent[]
     */
    public static function fromBitFlag(int $bitFlags): array
```